src/uct/api/v2/uct_v2.h

```diff
@@ -308,6 +319,36 @@ struct uct_ep_attr {
 typedef void (*uct_md_mem_invalidate_cb_t)(void *arg);
 
 
+/**
+ * @ingroup UCT_MD
+ * @brief Operation parameters passed to @ref uct_md_mem_reg_v2.
+ */
+typedef struct uct_md_mem_reg_params {
+    /**
+     * Mask of valid fields in this structure and operation flags, using
+     * bits from @ref uct_md_mem_reg_field_mask_t. Fields not specified
+     * in this mask will be ignored. Provides ABI compatibility with respect
+     * to adding new fields.
+     */
+    uint64_t                     field_mask;
+
+    /**
+     * Operation specific flags, using bits from @ref uct_md_mem_flags_t.
+     */
+    unsigned                     flags;
+
+    /**
+     * Memory to register.
+     */
+    void                         *address;
+
+    /**
+     * Size of memory to register. Must be > 0.
```

Comment:
>=0 ?

Reply:
it should be > 0, see https://github.com/openucx/ucx/blob/e9abd8ab7b1146b47a326954fc0d09d91992c943/src/uct/base/uct_md.c#L470
[ibv_reg_mr](https://www.rdmamojo.com/2012/09/07/ibv_reg_mr/) requires the following:
`Size of the memory block to register, in bytes. This value must be at least 1 and less than dev_cap.max_mr_size`

Reply:
@dmitrygx since length cannot be 0, and address cannot be NULL, does it mean address and length are mandatory fields? so they should be passed as regular parameters

Reply:
Do we treat the length zero as a special case because of a decade old bug in verbs registration code ?

Reply:
> Do we treat the length zero as a special case because of a decade old bug in verbs registration code ?

actually, this behavior came from `uct_mem_reg` API.
btw, registering zero-length doesn't sound as a thing that really makes sense.

Reply:
@yosefe @shamisp making memh to be some `NULL_HANDLE` stub or just `NULL` requires changing many places to support `NULL_HANDLE`/`NULL`.
Do you think it is worth it? IMO, it complicates the code.

Reply:
if this overcomplicates the code, I'm okay to stay with the requirement `(length!=0)`

Reply:
@shamisp yes, it over complicated the functionality.
see https://github.com/openucx/ucx/pull/8250#issuecomment-1139459009 for more details

Reply:
@dmitrygx I saw this comment. As I mentioned, I'm okay with not supporting length 0